mono/metadata/jit-icall-reg.h

```diff
@@ -343,11 +343,7 @@ MONO_JIT_ICALL (count) \
 
 #define MONO_JIT_ICALL_mono_get_lmf_addr MONO_JIT_ICALL_mono_tls_get_lmf_addr
 
-#ifdef __cplusplus
-typedef enum MonoJitICallId : gsize // Widen to gsize for use in MonoJumpInfo union.
-#else
```

Comment:
Can't it be enum MonoJitICallId : gsize everywhere ?

Reply:
Yes. The real problem is that mini-llvm.c isn't getting compiled as C++ even with configure -enable-cxx.

I can fix that, but it is a little risky, of accidental C compilation, until/unless the tree is unconditionally C++.

Reply:
I suppose you could argue, there isn't really any risk at present, since configure -enable-cxx doesn't have to work anyway.

Reply:
https://github.com/mono/mono/pull/17558 should be the correct fix.
But this `#if` is kinda a bomb waiting to go off as long as things are configurable.

Reply:
Note that I believe `enum foo : type` is not in C99 -- if that is what you mean by "everywhere".
If all the C compilers implement that, then yes, good.